tests/cpp/include/test_mkldnn.h

```diff
@@ -63,24 +63,24 @@ struct TestArrayShapes {
 };
 
 // Init arrays with the default layout.
-inline static void InitDefaultArray(NDArray *arr, bool is_rand = false) {
+inline static void InitDefaultArray(NDArray *arr, bool is_rand = false, int max = 50) {
   const TBlob &blob = arr->data();
   mshadow::default_real_t *data = blob.dptr<mshadow::default_real_t>();
   int size = blob.Size();
 
   for (int i = 0; i < size; i++)
     if (is_rand) {
-      data[i] = (std::rand() % 100) - 50;
+      data[i] = (std::rand() % (max * 2)) - max;
     } else {
-      data[i] = i % 100 - 50;
+      data[i] = i % (max * 2) - max;
```

Comment:
Same as above, how about change to something like `data[i] = i * 2.0 / size - 1.0` to generate [-1.0, 1.0)?